src/App.tsx

```diff
@@ -1,14 +1,61 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { getTodos } from './api';
+import { FilterStatusType, Todo } from './types/Todo';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+function filterFunction(
+  items: Todo[],
+  str: string,
+  filterStates: FilterStatusType,
+) {
+  let itemsInUse = [...items];
+
+  itemsInUse = itemsInUse.filter(todo =>
+    todo.title.toLowerCase().includes(str.toLowerCase().trim()),
+  );
+
+  itemsInUse = itemsInUse.filter(item => {
+    switch (filterStates) {
+      case FilterStatusType.All:
+        return true;
+      case FilterStatusType.Active:
+        return !item.completed;
+      case FilterStatusType.Completed:
+        return item.completed;
+      default:
+        return true;
+    }
+  });
+
+  return itemsInUse;
+}
+
 export const App: React.FC = () => {
+  const [todo, setTodo] = useState<Todo[]>([]);
+  const [filterLetter, setFilterLetter] = useState('');
+  const [filterStates, setFilterStatus] = useState<FilterStatusType>(
+    FilterStatusType.All,
+  );
+  const [loading, setLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const todoList = filterFunction(todo, filterLetter, filterStates);
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(todoFromServer => {
+        setTodo(todoFromServer);
+      })
```

Comment:
```suggestion
    getTodos()
      .then(setTodo)
```
I think Artur meant to pass the set function directly, it will work. Regarding `finally` all is ok with your variant